### PR TITLE
formatting fixes and code cleanup

### PR DESCRIPTION
**Pull request by JeffBezanson**

### CODE REVIEW

File: src/ccall.cpp [hunk 1]
```diff
@@ -1270,21 +1270,21 @@ static jl_cgval_t emit_ccall(jl_value_t **args, size_t nargs, jl_codectx_t *ctx)
                 if (jl_is_expr(args[2])) {
                     jl_expr_t *rtexpr = (jl_expr_t*)args[2];
                     if (rtexpr->head == call_sym && jl_expr_nargs(rtexpr) == 4 &&
-                            static_eval(jl_exprarg(rtexpr, 0), ctx, true, false) == jl_builtin_apply_type &&
-                            static_eval(jl_exprarg(rtexpr, 1), ctx, true, false) == (jl_value_t*)jl_array_type) {
+                        static_eval(jl_exprarg(rtexpr, 0), ctx, true, false) == jl_builtin_apply_type &&
+                        static_eval(jl_exprarg(rtexpr, 1), ctx, true, false) == (jl_value_t*)jl_array_type) {
```
Comment (tkelman) [inline src/ccall.cpp]: I liked this the way it was, following the matlab autoindenter's convention of giving line continuations of block openers (conditionals etc) an extra level of indent. The way you have it here, the conditional and the body are at the same indent so it's harder to parse where the conditional ends and the body starts.

Comment (JeffBezanson) [inline src/ccall.cpp]: Well, this occurs many places in the codebase. I'm just being consistent.

**Comment by eschnett** [comment]
If you are changing the formatting, have you considered using `clang-format`? That's a high quality tool, and if you're touching many files anyway, we might as well automate the formatting so that we won't have to think about it in the future.

Personally, I'm using `clang-format` for many projects -- I just run it automatically. It's amazing how much brain power is freed if you don't have to take care of pushing the space bar and enter key in just the right places.

File: src/ccall.cpp (hunk 1 above)
Comment (tkelman) [inline src/ccall.cpp]: the elseif's are indented in a clearer way than the ifs with this. it's pretty divided, we should probably pick a single preference.

**Comment by tkelman** [comment]
@eschnett can you revisit your attempts to configure clang format in a way that's mostly consistent with what we've been doing?

**Comment by StefanKarpinski** [comment]
It doesn't have to be perfectly consistent either. If we get it close enough, we can just change the formatting all at once.

**Comment by eschnett** [comment]
@StefanKarpinski I updated #15591.

The major differences between this and the current style are:
- preprocessor pragmas are not indented
- there is always a space before a pointer asterisk (`int *`)

There are many other differences, but I can't really tell whether these are just correcting local inconsistencies.

**Pull request merged**